plugins/generator-1.20.4/neoforge-1.20.4/templates/elementinits/items.java.ftl

```diff
@@ -41,13 +41,15 @@ package ${package}.init;
 <#assign hasDoubleBlocks = false>
 <#assign hasItemsWithProperties = w.getGElementsOfType("item")?filter(e -> e.customProperties?has_content)?size != 0
 	|| w.getGElementsOfType("tool")?filter(e -> e.toolType == "Shield")?size != 0>
+<#assign itemsWithInventory = w.getGElementsOfType("item")?filter(e -> e.hasInventory())>
 
-<#if hasItemsWithProperties>
-@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD, value = Dist.CLIENT)
+<#if hasItemsWithProperties || itemsWithInventory?size != 0>
+@Mod.EventBusSubscriber(bus = Mod.EventBusSubscriber.Bus.MOD)
 </#if>
 public class ${JavaModName}Items {
 
 	public static final DeferredRegister<Item> REGISTRY = DeferredRegister.create(BuiltInRegistries.ITEM, ${JavaModName}.MODID);
+	public static final DeferredRegister<AttachmentType<?>> ATTACHMENT_TYPES = DeferredRegister.create(NeoForgeRegistries.Keys.ATTACHMENT_TYPES, ${JavaModName}.MODID);

```

Comment:
This one is not always needed, make it conditional

Reply:
@Defeatomizer when I wrote this, I did not pay attention it is also used referenced in main mod file so both options work.

One produces cleaner code for the user, the other one is faster as there is no filter call in main mod file

Reply:
Mark for RFR when you think so

Reply:
So better keep it as it was (to follow variables class)?

Reply:
I have an alternative solution that fixes both. Make a new static method in items elementinit file called

`${JavaModName}Items.registerRegistries(IEventBus)` that registers the files needed.

Reply:
:heavy_check_mark: Fixed